### PR TITLE
add ts measures

### PR DESCRIPTION
**Pull request by 0xMuluh**
ping: #89 
Added wrapper for acf, pacf, arima and box.test

### CODE REVIEW

File: R/getTsMeasure.R [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        }
+        
+        ############################ Input check end ###########################
+        
+        ############################ Measure Calculation ######################
+        # Initialize a list to store results
+        results <- list()
+        
+        # Loop through the specified measures and calculate each
+        for (measure in measures) {
```
Comment (TuomasBorman) [inline R/getTsMeasure.R]: Use lapply instead

File: R/getTsMeasure.R [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+            if (measure == "acf") {
+                results$acf <- acf(ts_data, plot = FALSE)$acf
```
Comment (TuomasBorman) [inline R/getTsMeasure.R]: The calculation part could be moved to internal function

Comment (TuomasBorman) [reply R/getTsMeasure.R]: Also try to use switch() whereever possible (or other that kind of structure). It makes the code cleaner when the calculation is done in one step

Consider this kind of structure:
```
exportedFUN <- function(){

res <- lapply(methods, function(method){
.internal_fun(method)
}
names(res) <- measures
return(res)
}


.internal_fun <- function(measure){
# Get correct function to call
FUN <- switch()
# Calculate
res <- FUN()
# Get the result
if( measure %in% c("acf", "pacf") ){
res <- res$acf
}
return(res)
}
```

File: R/getTsMeasure.R (hunk 1 above)
Comment (0xMuluh) [reply R/getTsMeasure.R]: ACK

File: R/getTsMeasure.R (hunk 2 above)
Comment (0xMuluh) [reply R/getTsMeasure.R]: ACK

**Comment by 0xMuluh** [comment]
@TuomasBorman @antagomir

File: R/getTsMeasure.R [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+#' @export
+setGeneric("getTsMeasure", signature = "x",
+    function(x, ...) 
+       standardGeneric("getTsMeasure")
+)
+
+#' @rdname getTsMeasure
+#' @export
+setMethod("getTsMeasure", signature = c(x = "TreeSummarizedExperiment"),
+    function(x, time.field = "Time.hr", measures = "acf", ...) {
```
Comment (TuomasBorman) [inline R/getTsMeasure.R:33]: As time.field can be anything, I don't believe there is benefits by setting the default value. So time.field could not have any value bu default. That might be more clear

File: R/getTsMeasure.R [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+#' from a specified column in the `colData` of a `TreeSE` object.
+#' 
+#' @param x A
+#' \code{\link[SummarizedExperiment:SummarizedExperiment-class]{SummarizedExperiment}}
+#' object.
+#' @param time.field \code{Character scalar}. Specifies the name of the
+#' time series field in `colData`. (Default: \code{"Time.hr"})
+#' @param measures \code{Character vector}. Specifies the time series measures 
+#' to compute. Supported measures include "acf", "pacf","Box.test", and "arima".
+#' (Default: \code{ ""acf""})
```
Comment (TuomasBorman) [inline R/getTsMeasure.R:12]: ""acf"" --> "acf"

File: R/getTsMeasure.R [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+                data.", call. = FALSE)
+    }
+    
+    # Get the correct ts measure to call
+    FUN <- switch(measure,
+                  "acf" = function() acf(ts_data, plot = FALSE),
+                  "pacf" = function() pacf(ts_data, plot = FALSE),
+                  "Box.test" = function() Box.test(ts_data, type = "Ljung-Box"),
+                  "arima" = function() arima(ts_data),
+                  stop("Unsupported measure: ", measure, call. = FALSE))
```
Comment (TuomasBorman) [inline R/getTsMeasure.R:75]: Pass additional arguments with "..."

File: R/getTsMeasure.R (hunk 4 above)
Comment (TuomasBorman) [inline R/getTsMeasure.R:12]: Documentation is lacking info. For instance, arima model has lots of parameters that user wants to modify. However, now it is very hard to know what are those parameter names

**Review by TuomasBorman — CHANGES_REQUESTED** [review]
Unfortunately, I’ve only had time to review this function thoroughly now, and it seems that it's not working correctly. The issue is that it calculates measures/models for time point values(?), which isn't meaningful. Instead, it should calculate the values based on abundances. Additionally, there needs to be a way to group values based on patient.

It seems that the functions you are utilizing expect the input format to have rows representing time points and columns representing patients.
To fix this:

1. Modify the abundance table so that rows represent time points and columns represent patients.
2. Check if missing time point values are handled by the functions by default.
3. Utilize the functions. Pass additional parameters using `...`
5. Improve the documentation, as it doesn’t currently describe which functions are used. Each of these methods has a unique set of parameters that need to be clarified.

**Comment by antagomir** [comment]
Thanks. I suggest that we hold dealing with this PR until other critical issues have been dealt with. This is extra functionality which is not urgent.

**Comment by antagomir** [comment]
I will close this now. Analysis and visualization functionality should be kept separate. We can reopen this when the need for ts measure functionality is clarified. 

We could discuss more in #89 before deciding *how* to implement this.

**Comment by antagomir** [comment]
Or alternatively, we can limit the number of measures to few that are easy to handle well, if this PR was close to be ready.

**Pull request closed without merging**